user_sync/config.py

```diff
@@ -497,7 +497,19 @@ def get_rule_options(self):
 
         # get the limits
         limits_config = self.main_config.get_dict_config('limits')
-        options['max_adobe_only_users'] = limits_config.get_int('max_adobe_only_users')
+        max_missing = limits_config.get_value('max_adobe_only_users',(int, str),False)
+        if isinstance(max_missing, str):
+            percent_pattern = re.compile("(\d*(\.\d+)?%)")
+            if percent_pattern.match(max_missing):
+                max_missing_percent = float(max_missing.strip('%'))
+                if max_missing_percent >= 0.0 and max_missing_percent <= 100.0:
+                    options['max_adobe_only_users'] = max_missing
+                else:
+                    raise AssertionException("max_adobe_only_users value must be less or equal than 100%")
+            else:
+                raise AssertionException("Not a valid value for max_adobe_only_users. Must be a percentage or fix integer")
+        else:
+            options['max_adobe_only_users'] = max_missing
```

Comment:
I get an error when I set `max_adobe_only_users` to an integer as a string.  e.g.

```yaml
  max_adobe_only_users: "250"
```

We need to be able to specify the number in this manner to preserve backward compatibility.  I think you could also simplify your validation steps:

1. If `max_adobe_only_users` is a string and contains a "%" symbol, validate with the regex and check percentage bounds.
2. Otherwise cast the value `max_adobe_only_users` to an int and if you catch a ValueError, raise an ArgumentException that the value must be an integer.  note: you may need to wrap the ValueError check in a helper function that returns a bool.

```python
try:
  int(max_missing)
except ValueError:
  # raise or return from helper func
```